### ISSUE TITLE
Improve the display of variables in debugger

### ISSUE THREAD

**Issue by lukehoban** [Microsoft/vscode-go#7]
There are a number of things to do here:
- [X] Don't render separate child for pointer dereference (https://github.com/Microsoft/vscode-go/commit/598b92cb9543afb37a38c81dd9cf2068aca0c5f7)
- [X] Show string values with quotes (https://github.com/Microsoft/vscode-go/commit/598b92cb9543afb37a38c81dd9cf2068aca0c5f7)
- [X] Render arrays correctly (https://github.com/Microsoft/vscode-go/commit/598b92cb9543afb37a38c81dd9cf2068aca0c5f7)
- [x] Render maps correctly
- [x] Allow following members indefinitely (not limited by Delve default depths)
- [X] Render objects as explorable trees in Watch and Debug Console windows (9e1edf70b5680b791bdf55c7cb8478a90d788fcf) 
- [X] Send `ThreadEvent`s so that threads window is accurate (https://github.com/Microsoft/vscode/issues/40)  (e51037aea4ca0d3a2a4ec22344e5213e58375051)

**Comment by ae6rt** [comment]
Would adding 

```
* Show package variables
```

qualify to be on this list?  If I set a watch on a package variable *main.foo*, I can see the value, but I cannot find a way to display a given package's variables without explicitly asking for each one.

**Comment by ramya-rao-a** [comment]
Checked off the item for maps as we got that done via #1384
Checked off the item for following members and not limited by dlv's default depth as we can now configure the depth to which delve goes when resolving variables. See https://github.com/Microsoft/vscode-go/issues/1804#issuecomment-408654035

@ae6rt We can only display what delve allows us to. dlv allows us to see all local variables and so we have the pane that shows the same. I dont see how we can show all package variables without asking for each.
